### PR TITLE
feat: Bundle optimization and first load improvements

### PR DESCRIPTION
**Pull request by SatishGandham**
## Description
This is an intermediate branch for the first load optimization

### TL;DR 
Improves the first load times via bundle optimization, code splitting, and prioritizing delivery of critical code.

### Change 1
The Evaluation and Tern workers have several large dependencies:

- linkedom/worker
- node-forge
- tern JSON definitions
This PR removes them from the main bundle. When applied to appsmithorg/appsmith on top of https://github.com/appsmithorg/design-system/pull/389, this makes the bundle 980 minified kB smaller (7.46 → 6.48 minified MB). #20972 

### Change 2

- switches the build target to modern browsers; this means stuff like class, async/await, etc is not transpiled anymore
- code-splits away the code editor + sets up ESLint rules to enforce that
- code-splits away Lottie + sets up ESLint rules to enforce that

In total, this saves 1.34 minified MBs (8.34 → 7 MB), getting rid of these modules (+ making others smaller, thanks to the modern build target):

Fixes # (issue)
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change
First load optimization and some housekeeping stuff.

### Test Plan
- Test icons in icon picker and rest of the application. Check if icons are loading properly and they don't have intermedite blank state.
- Test code editor how it loads are there any delays in interacting with it, use network throttling when testing.

 

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by chandannkumar** [comment]
No DP available in this PR
@SatishGandham can you please check

File: app/client/cypress/support/ApiCommands.js [hunk 1]
```diff
@@ -146,6 +146,7 @@ Cypress.Commands.add(
 );
 
 Cypress.Commands.add("EditSourceDetail", (baseUrl, v1method) => {
+  cy.EnableAllCodeEditors();
```
[the remark this answers is not in the document]
Comment (iamakulov) [reply app/client/cypress/support/ApiCommands.js:149]: Previously, only select code editors were lazy (i.e., were loaded and rendered on-demand). The one this command works with didn’t.

Now, all code editors are lazy. They first render as a `<pre>` at first – and only then, when the editor loads and `requestIdleCallback()` fires, re-render as genuine `CodeMirror`s.

This means we now need to check that the code editor initialized before any interaction with it.

**Comment by chandannkumar** [comment]
Observed the following:
- change of icons style of autocomplete list of code editor (as discussed with @SatishGandham )
Screenshot (265)

- More icons got added in icon lists (Compared with current release version)
Screenshot (267)

**Comment by chandannkumar** [comment]
Tested this PR and LGTM
- No breakage observed
- No impact on existing widgets or their working functionalities

**Comment by somangshu** [comment]
Build running [here](https://github.com/appsmithorg/appsmith/actions/runs/4936274643)

**Comment by iamenrique** [comment]
After merging this PR, is everything in this guide still valid? https://github.com/appsmithorg/appsmith/blob/release/contributions/ClientSetup.md

I noticed node_modules and ESLint configuration are now in a different level. Does this mean the `yarn install` has to be done in the `app/` folder rather than `app/client`? Also had to change the ESLint configuration manually in IntelliJ (was automatic before).

**Comment by SatishGandham** [comment]
You can do yarn install in any folder under app. 

We will move all UI related code under client in a couple of weeks.

**Comment by iamenrique** [comment]
@SatishGandham Understood. Is there any issue number / branch / PR I can follow to be aware of those updates? Thanks in advance.

**Comment by SatishGandham** [comment]
@iamenrique , https://github.com/appsmithorg/appsmith/pull/23395/
